File: public/.julia/v0.5/mosek/deps/src/mosek/8/tools/examples/fusion/python/lo1.py
##
#  Copyright: Copyright (c) MOSEK ApS, Denmark. All rights reserved.
#
#  File:      lo1.py
#
#  Purpose: Demonstrates how to solve the problem
#
#  minimize 3*x1 + 5*x2 + x3 + y
#  such that
#           3*x1 + 2*x2        +   y = 30,
#           2*x1 + 3*x2 +   x3 +   y > 15,
#                       + 3*x3 + 2*y < 25
#  and
#           x0,x1,x2 => 0,
#           0 <= y <= 10
##

import sys

import mosek
import mosek.fusion
from  mosek.fusion import *

def main(args):
    A = [ [ 3.0, 2.0, 0.0, 1.0 ],
          [ 2.0, 3.0, 1.0, 1.0 ],
          [ 0.0, 0.0, 3.0, 2.0 ] ]
    c = [ 3.0, 5.0, 1.0, 1.0  ]

    # Create a model with the name 'lo1'
    with Model("lo1") as M:

      # Create variable 'x' of length 3
      x = M.variable("x", 4, Domain.greaterThan(0.0))

      # Create three constraints
      M.constraint("c1", Expr.dot(A[0], x), Domain.equalsTo(30.0))
      M.constraint("c2", Expr.dot(A[1], x), Domain.greaterThan(15.0))
      M.constraint("c3", Expr.dot(A[2], x), Domain.lessThan(25.0))

      # Set the objective function to (c^t * x)
      M.objective("obj", ObjectiveSense.Maximize, Expr.dot(c, x))
        
      # Solve the problem
      M.solve()


      # Get the solution values
      sol = x.level()
      print('\n'.join(["x[%d] = %f"%(i, sol[i]) for i in range(4)]) )

if __name__ == '__main__':
    main(sys.argv[1:])

File: public/.julia/v0.5/mosek/deps/src/mosek/8/tools/examples/fusion/cxx/lo1.cc
////
//  Copyright: Copyright (c) MOSEK ApS, Denmark. All rights reserved.
//
//  File:      lo1.cc
//
//  Purpose: Demonstrates how to solve the problem
//
//  minimize 3*x0 + 5*x1 + x2 + x3
//  such that
//           3*x0 + 2*x1        +   x3 = 30,
//           2*x0 + 3*x1 +   x2 +   x3 > 15,
//                       + 3*x2 + 2*x3 < 25
//  and
//           x0,x1,x2 > 0,
//           0 < x3 < 10
////

#include <memory>
#include <iostream>


#include "fusion.h"

using namespace mosek::fusion;
using namespace monty;

int main(int argc, char ** argv)
{
  auto A1 = new_array_ptr<double,1>({ 3.0, 2.0, 0.0, 1.0 });
  auto A2 = new_array_ptr<double,1>({ 2.0, 3.0, 1.0, 1.0 });
  auto A3 = new_array_ptr<double,1>({ 0.0, 0.0, 3.0, 2.0 });
  auto c  = new_array_ptr<double,1>({ 3.0, 5.0, 1.0, 1.0 });

  Model::t M = new Model("lo1"); auto _M = finally([&]() { M->dispose(); });

  M->setLogHandler([=](const std::string & msg) { std::cout << msg << std::flush; } );

  Variable::t x = M->variable("x", 4, Domain::greaterThan(0.0));

  M->constraint("c1", Expr::dot(A1, x), Domain::equalsTo(30.0));
  M->constraint("c2", Expr::dot(A2, x), Domain::greaterThan(15.0));
  M->constraint("c3", Expr::dot(A3, x), Domain::lessThan(25.0));

  M->objective("obj", ObjectiveSense::Maximize, Expr::dot(c, x));

  M->solve();

  auto sol = x->level();
  
  std::cout << "[x0,x1,x2,x3] = [" << (*sol)[0] << "," << (*sol)[1] << "," << (*sol)[2] << " ]\n";
}
